spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/LazySpanSubscriber.java

```diff
@@ -29,6 +29,7 @@
  * @author Marcin Grzejszczak
  * @since 2.0.0
  */
+// TODO: why are we extending AtomicBoolean and not actually using its methods?
```

Comment:
Indeed, if the CAS methods are not used then extending `AtomicBoolean` is not useful. Have you looked at `LazySpanSubscriber` instances being passed around / used as `AtomicBoolean`s maybe?

Reply:
we think this was copy/paste error and nothing more. good idea (checked.)